doc/proposals/scorecard-user-tests.md

```diff
@@ -0,0 +1,153 @@
+# User Defined Tests for the Operator Scorecard
+
+Implementation Owner: AlexNPavel
+
+Status: Draft
+
+[Background](#Background)
+
+[Goals](#Goals)
+
+[Design overview](#Design_overview)
+
+[User facing usage](#User_facing_usage)
+
+[Observations and open questions](#Observations_and_open_questions)
+
+## Background
+
+The operator scorecard is intended to allow users to run a generic set of tests on their operators. Some simple checks can be performed, but more complicated
+tests verifying that the operator actually works are not possible to do in a generic way. This leads to some of the functional tests in the current scorecard
+implementation to be inaccurate or too insufficient to be useful. For more useful functional scorecard tests, we need to allow some basic user input for tests.
+
+## Goals
+
+- Implement user-defined scorecard tests
+- Replace existing "Operator actions are reflected in status" and "Writing into CRs has effects" tests with user-defined tests
+
+## Design overview
+
+### Basic YAML Defined Test
+
+A new basic testing system would be added where a user can simply define various aspects of a test. For example, this definition runs a similar test to the memcached-operator scale test from the SDK's e2e test:
```

Comment:
The yaml file below would need to be in a predefined directory or passed to the tests?

Reply:
The config path would be set via an environment variable (see the user_defined_tests config example below)